include/nbl/builtin/glsl/ext/DepthPyramidGenerator/depth_pyramid_generator_impl.glsl

```diff
@@ -0,0 +1,87 @@
+
+#ifdef REDUCION_OP_MIN
+#define REDUCTION_OPERATOR min
+#define REDUCTION_OPERATOR_2 min
+#define REDUCED_VAL_T float
+#elif defined(REDUCION_OP_MAX)
+#define REDUCTION_OPERATOR max
+#define REDUCTION_OPERATOR_2 max
+#define REDUCED_VAL_T float
+#elif defined(REDUCION_OP_BOTH)
+#define REDUCTION_OPERATOR(a, b) vec2(min(a, b), max(a, b))
+#define REDUCTION_OPERATOR_2(a, b) vec2(min(a.x, b.x), max(a.x, b.x))
+#define REDUCED_VAL_T vec2
+#endif
+
+layout(binding = 0, set = 0) uniform sampler2D sourceTexture;
+layout(binding = 1, set = 0, MIP_IMAGE_FORMAT) uniform image2D outMips[MIPMAP_LEVELS_PER_PASS];
+
+void storeReducedVal(in uint mipIdx, in uvec2 coords, float reducedVal)
+{
+  imageStore(outMips[mipIdx], ivec2(coords), vec4(reducedVal, 0.f, 0.f, 0.f));
+}
+
+void storeReducedVal(in uint mipIdx, in uvec2 coords, vec2 reducedVal)
+{
+  imageStore(outMips[mipIdx], ivec2(coords), vec4(reducedVal, 0.f, 0.f));
+}
+
+#include "nbl/builtin/glsl/utils/morton.glsl"
+
+#define WORKGROUP_SIZE (WORKGROUP_X_AND_Y_SIZE * WORKGROUP_X_AND_Y_SIZE)

```

Comment:
standardised define `NBL_GLSL_WORKGROUP_SIZE` and the LOG2 value (or whatever its called in other builtin shaders)